core/services/ocrbootstrap/delegate.go

```diff
@@ -118,14 +118,15 @@ func (d *Delegate) ServicesForSpec(jobSpec job.Job, qopts ...pg.QOpt) (services
 		if routerFields.ContractVersion != 1 || routerFields.ContractUpdateCheckFrequencySec == 0 {
 			return nil, errors.New("invalid router contract config")
 		}
-		configProvider, err = relayer.NewFunctionsProvider(
+		configProvider, err = relayer.NewPluginProvider(
 			ctx,
 			types.RelayArgs{
 				ExternalJobID: jobSpec.ExternalJobID,
 				JobID:         spec.ID,
 				ContractID:    spec.ContractID,
 				RelayConfig:   spec.RelayConfig.Bytes(),
 				New:           d.isNewlyCreatedJob,
+				ProviderType:  string(types.Functions),
```

Comment:
do we need the type check here that you have for median? if not, why not?

Reply:
We don't because NewPluginProvider returns a PluginProvider, which is a strict superset of ConfigProvider (it adds the ContractTransmitter). Therefore, no type coercion needed.